_posts/2022-04-04-valheim-server.md

```diff
@@ -0,0 +1,130 @@
+- [Visual Studio Code](https://code.visualstudio.com) installed
+- [Node.js](https://nodejs.org/en/download/) installed
+- [AWS Command Line Interface](https://aws.amazon.com/cli/) installed
+- [AWS SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) installed
+- Following github repo downloaded [Valheim Server](https://github.com/JMiocevich/valheim_server_aws_ts)
+- [Discord WebHook](https://support.discord.com/hc/en-us/articles/228383668-Intro-to-Webhooks)
+
+See [ setup instructions ](https://github.com/JMiocevich/Valheim-Server-on-AWS-Part-1/blob/main/README.md) for step by step instructions on setup.
+
+## Discussion and Reasoning
+
+### Why Fargate?
+Takes advantage of [valheim-server docker](ht
+tps://github.com/lloesche/valheim-server-docker), allows for simple deployment without having to manage servers.
+
+### Why spot Instance?
+It's a game and is not a critical workload, and so far have not found it to be any problem in availability. Also $$$.
+
+### Why Lambda Functions for server start/stop
+In future, a discord command/HTTP endpoint will be implemented to invoke this lambda, allowing for an easy server start, currently, it needs to be invoked manually via the CLI or console. The stop function allows for cloud watch to trigger an SNS topic, which in turn invokes the lambda stop function.
+
+
+### Why discord notification?
+Allows for easy monitoring of Server Status and a player connects to the server.
+
+![Valheim Discord Notifications](/img/valheim_discord.png)
+<br/>
+
+Instructions on how to do this are in the [ lloesche valheim-server](https://github.com/lloesche/valheim-server-docker) repo.
+
+<br/>
+
+### For the Cloudwatch Alarm, why use CPU Usage as a metric to track server usage?
+Ideally, I would like to parse the logs/ or ping the server for player count, however, I was having trouble getting it to work reliably. In addition, this ensures that the server will not shut down if an update or backup is been actioned. 
+
+### Why no static IP?
+If the game is set to public: true, the server can be searched for using the game server browser. Save some $$$.
+
+### Secret Manager, but why?
+Good practise, Lord Myerscough will sense a disturbance in the code...be afraid.
+
+
+### Cost. 
+[https://aws.amazon.com/fargate/pricing/](https://aws.amazon.com/fargate/pricing/) \
+[https://aws.amazon.com/s3/pricing/](https://aws.amazon.com/s3/pricing/) 
+
+
+At the time of writing with 24-hour usage: \
+$0.118 per hour spot instance at the time of writing \
+$2.84 per day spot instance at the time of writing \
+~$80 per month 
+
+However this is with 24-hour usage, with auto-scale down this cost reduces significantly, and you only pay for what you use. The cloud watch alarm is set to alarm if CPU usage drops below a threshold of 25/30 with a period of 60 seconds.
```

Comment:
What has typical cost been for you?  And how does that compare to other options you mentioned in the introduction.  This is good to frame your solution in relation to the original requirements and the alternatives.

Reply:
Will add in cost comparisons